diff -rup 3.2.10.orig/HTML/QuickForm/advcheckbox.php 3.2.10/HTML/QuickForm/advcheckbox.php
--- 3.2.10.orig/HTML/QuickForm/advcheckbox.php	2007-10-05 22:29:00.000000000 +0200
+++ 3.2.10/HTML/QuickForm/advcheckbox.php	2008-11-17 15:53:10.000000000 +0100
@@ -249,7 +249,10 @@ class HTML_QuickForm_advcheckbox extends
                 // default values are overriden by submitted
                 $value = $this->_findValue($caller->_constantValues);
                 if (null === $value) {
-                    $value = $this->_findValue($caller->_submitValues);
+					if($this->_flagFrozen) 
+						$this->_removeValue($caller->_submitValues);
+					else
+	                    $value = $this->_findValue($caller->_submitValues);
                     if (null === $value) {
                         $value = $this->_findValue($caller->_defaultValues);
                     }
diff -rup 3.2.10.orig/HTML/QuickForm/checkbox.php 3.2.10/HTML/QuickForm/checkbox.php
--- 3.2.10.orig/HTML/QuickForm/checkbox.php	2007-10-05 22:29:00.000000000 +0200
+++ 3.2.10/HTML/QuickForm/checkbox.php	2008-11-17 15:53:10.000000000 +0100
@@ -146,10 +146,10 @@ class HTML_QuickForm_checkbox extends HT
     function getFrozenHtml()
     {
         if ($this->getChecked()) {
-            return '<tt>[x]</tt>' .
+            return '<div class="checkbox_on"></div>' .
                    $this->_getPersistantData();
         } else {
-            return '<tt>[ ]</tt>';
+            return '<div class="checkbox_off"></div>';
         }
     } //end func getFrozenHtml
 
@@ -239,7 +239,11 @@ class HTML_QuickForm_checkbox extends HT
                     // if no boxes were checked, then there is no value in the array
                     // yet we don't want to display default value in this case
                     if ($caller->isSubmitted()) {
-                        $value = $this->_findValue($caller->_submitValues);
+						if($this->_flagFrozen) {
+							$this->_removeValue($caller->_submitValues);
+							$value = $this->_findValue($caller->_defaultValues);
+						} else
+            	            $value = $this->_findValue($caller->_submitValues);
                     } else {
                         $value = $this->_findValue($caller->_defaultValues);
                     }
diff -rup 3.2.10.orig/HTML/QuickForm/element.php 3.2.10/HTML/QuickForm/element.php
--- 3.2.10.orig/HTML/QuickForm/element.php	2007-10-05 22:29:00.000000000 +0200
+++ 3.2.10/HTML/QuickForm/element.php	2008-11-17 15:57:37.000000000 +0100
@@ -76,6 +76,14 @@ class HTML_QuickForm_element extends HTM
      */
     var $_persistantFreeze = false;
     
+	/**
+	 * Used by freeze to remove submit values.
+     * @var       object
+     * @since     epesi
+     * @access    private
+	 */
+	var $_caller = null;
+	
     // }}}
     // {{{ constructor
     
@@ -205,6 +213,10 @@ class HTML_QuickForm_element extends HTM
     function freeze()
     {
         $this->_flagFrozen = true;
+		if($this->_caller!==null) {
+			$this->_removeValue($this->_caller->_submitValues);
+			$this->onQuickFormEvent('updateValue',null,$this->_caller);
+		}
     } //end func freeze
 
     // }}}
@@ -353,6 +365,30 @@ class HTML_QuickForm_element extends HTM
             return null;
         }
     } //end func _findValue
+
+    /**
+     * Tries to remove the element value from the values array
+     * 
+     * @since     2.7
+     * @access    private
+     * @return    mixed
+     */
+    function _removeValue(&$values)
+    {
+        if (empty($values)) {
+            return null;
+        }
+        $elementName = $this->getName();
+        if (isset($values[$elementName])) {
+            unset($values[$elementName]);
+        } elseif (strpos($elementName, '[')) {
+            $myVar = "['" . str_replace(
+                         array('\\', '\'', ']', '['), array('\\\\', '\\\'', '', "']['"), 
+                         $elementName
+                     ) . "']";
+            eval("if(isset(\$values$myVar)) unset(\$values$myVar);");
+        } 
+    } //end func _findValue
 
     // }}}
     // {{{ onQuickFormEvent()
@@ -369,6 +405,7 @@ class HTML_QuickForm_element extends HTM
      */
     function onQuickFormEvent($event, $arg, &$caller)
     {
+		$this->_caller = & $caller;
         switch ($event) {
             case 'createElement':
                 $className = get_class($this);
@@ -383,7 +420,10 @@ class HTML_QuickForm_element extends HTM
                 // default values are overriden by submitted
                 $value = $this->_findValue($caller->_constantValues);
                 if (null === $value) {
-                    $value = $this->_findValue($caller->_submitValues);
+					if($this->_flagFrozen) 
+						$this->_removeValue($caller->_submitValues);
+					else
+	                    $value = $this->_findValue($caller->_submitValues);
                     if (null === $value) {
                         $value = $this->_findValue($caller->_defaultValues);
                     }
@@ -432,7 +472,7 @@ class HTML_QuickForm_element extends HTM
         static $idx = 1;
 
         if (!$this->getAttribute('id')) {
-            $this->updateAttributes(array('id' => 'qf_' . substr(md5(microtime() . $idx++), 0, 6)));
+            $this->updateAttributes(array('id' => 'qf_'.$idx));// . substr(md5($idx++), 0, 6)));
         }
     } // end func _generateId
 
diff -rup 3.2.10.orig/HTML/QuickForm/radio.php 3.2.10/HTML/QuickForm/radio.php
--- 3.2.10.orig/HTML/QuickForm/radio.php	2007-10-05 22:29:01.000000000 +0200
+++ 3.2.10/HTML/QuickForm/radio.php	2008-11-17 15:53:10.000000000 +0100
@@ -146,10 +146,10 @@ class HTML_QuickForm_radio extends HTML_
     function getFrozenHtml()
     {
         if ($this->getChecked()) {
-            return '<tt>(x)</tt>' .
+            return '<div class="radio_on"></div>' .
                    $this->_getPersistantData();
         } else {
-            return '<tt>( )</tt>';
+            return '<div class="radio_off"></div>';
         }
     } //end func getFrozenHtml
 
@@ -205,7 +205,10 @@ class HTML_QuickForm_radio extends HTML_
                 // default values are overriden by submitted
                 $value = $this->_findValue($caller->_constantValues);
                 if (null === $value) {
-                    $value = $this->_findValue($caller->_submitValues);
+					if($this->_flagFrozen) 
+						$this->_removeValue($caller->_submitValues);
+					else
+	                    $value = $this->_findValue($caller->_submitValues);
                     if (null === $value) {
                         $value = $this->_findValue($caller->_defaultValues);
                     }
Only in 3.2.10/HTML/QuickForm/Renderer: .svn
diff -rup 3.2.10.orig/HTML/QuickForm/Rule/Required.php 3.2.10/HTML/QuickForm/Rule/Required.php
--- 3.2.10.orig/HTML/QuickForm/Rule/Required.php	2007-10-05 22:29:00.000000000 +0200
+++ 3.2.10/HTML/QuickForm/Rule/Required.php	2008-11-08 19:32:39.000000000 +0100
@@ -47,6 +47,7 @@ class HTML_QuickForm_Rule_Required exten
      */
     function validate($value, $options = null)
     {
+    	if (is_string($value)) $value=trim($value);
         if ((string)$value == '') {
             return false;
         }
Only in 3.2.10/HTML/QuickForm/Rule: .svn
diff -rup 3.2.10.orig/HTML/QuickForm/select.php 3.2.10/HTML/QuickForm/select.php
--- 3.2.10.orig/HTML/QuickForm/select.php	2007-10-05 22:29:01.000000000 +0200
+++ 3.2.10/HTML/QuickForm/select.php	2008-11-24 11:30:43.000000000 +0100
@@ -593,10 +593,13 @@ class HTML_QuickForm_select extends HTML
         if ('updateValue' == $event) {
             $value = $this->_findValue($caller->_constantValues);
             if (null === $value) {
-                $value = $this->_findValue($caller->_submitValues);
+				if($this->_flagFrozen) 
+					$this->_removeValue($caller->_submitValues);
+				else
+	                $value = $this->_findValue($caller->_submitValues);
                 // Fix for bug #4465 & #5269
                 // XXX: should we push this to element::onQuickFormEvent()?
-                if (null === $value && (!$caller->isSubmitted() || !$this->getMultiple())) {
+                if (null === $value && ((is_callable(array($caller,'isSubmitted')) && !$caller->isSubmitted()) || (is_callable(array($this,'getMultiple')) && !$this->getMultiple()))) {
                     $value = $this->_findValue($caller->_defaultValues);
                 }
             }
Only in 3.2.10/HTML/QuickForm: .svn
diff -rup 3.2.10.orig/HTML/QuickForm.php 3.2.10/HTML/QuickForm.php
--- 3.2.10.orig/HTML/QuickForm.php	2007-10-05 22:29:01.000000000 +0200
+++ 3.2.10/HTML/QuickForm.php	2008-11-24 11:52:21.000000000 +0100
@@ -593,7 +593,9 @@ class HTML_QuickForm extends HTML_Common
         }
         $className = $GLOBALS['HTML_QUICKFORM_ELEMENT_TYPES'][$type][1];
         $includeFile = $GLOBALS['HTML_QUICKFORM_ELEMENT_TYPES'][$type][0];
-        include_once($includeFile);
+        static $cache=array();
+        if (!isset($cache[$includeFile])) include_once($includeFile);
+        $cache[$includeFile] = true;
         $elementObject =& new $className();
         for ($i = 0; $i < 5; $i++) {
             if (!isset($args[$i])) {
@@ -1512,6 +1514,7 @@ class HTML_QuickForm extends HTML_Common
         $registry =& HTML_QuickForm_RuleRegistry::singleton();
 
         foreach ($this->_rules as $target => $rules) {
+			if($this->isElementFrozen($target)) continue;
             $submitValue = $this->getSubmitValue($target);
 
             foreach ($rules as $rule) {
Only in 3.2.10/HTML: .svn
Only in 3.2.10: .svn
